### PR TITLE
all: fix grammar, typos, reformat tables

### PR DESCRIPTION
**Pull request by J4bbi**
### Description

This PR fixes obvious grammatical mistakes, typos and reformats some tables to make the markdown more readabe/correct.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Review by fenekku — COMMENTED** [review]
Thank you so much for this! This must have been a gradual investment over time and we really appreciate it. 

There is a couple of further adjustments I added. I will take another look after and then merge. 

This will really level up the professionalism of the project. Thanks so much again!

File: docs/reference/metadata.md [hunk 1]
```diff
@@ -83,15 +83,15 @@ A record stores information about two internal PIDs:
...
+|  Field  | Description                                                                   |
+|:-------:|:------------------------------------------------------------------------------|
+| ``id``  | The value of the internal record identifier.                                  |
 | ``pid`` | Object level information about the identifier needed for operational reasons. |
 
 Concept version PID:
 
-| Field |   Description   |
-|:-----:|:----------------|
+|     Field     | Description                                 |
```
[the remark this answers is not in the document]
Comment (J4bbi) [reply docs/reference/metadata.md:93]: I've left the centering, just because it's more faff to revert it if that's okay

**Comment by J4bbi** [comment]
@fenekku , thanks for the review, I've responded to all your comments, and I also changed a couple of other thing I noticed

I removed the last sentence 
"All of this can be further complicated if you run container images or virtual machines" from virtualenvs.md because I didn't think it added much value.

File: docs/develop/architecture/software.md [hunk 2]
```diff
@@ -24,13 +24,13 @@ The diagram below shows a simplified view of the data flow in the architecture.
 
 ![Architecture layers](../img/architecture.svg)
 
-*The presentation layer* parses incoming requests and routes them to service layer. This involves sending and receiving data in multiple different formats and translating these into an internal representation, as well as e.g. parsing arguments from an HTTP request (e.g parsing the query string parameters).
+*The presentation layer* parses incoming requests and routes them to service layer. This involves sending and receiving data in multiple different formats and translating these into an internal representation, as well as e.g., parsing arguments from an HTTP request (e.g., parsing the query string parameters).
```
Comment (fenekku) [inline docs/develop/architecture/software.md:27]:
```suggestion
*The presentation layer* parses incoming requests and routes them to service layer. This involves sending and receiving data in multiple different formats and translating these into an internal representation. For instance, this includes parsing arguments from an HTTP request e.g., parsing the query string parameters.
```

File: docs/develop/architecture/software.md [hunk 3]
```diff
@@ -24,13 +24,13 @@ The diagram below shows a simplified view of the data flow in the architecture.
...
 
 *The data access layer* is responsible for ensuring data integrity, harmonizing data access to different storages as well as fetching and storing the data in the underlying systems.
 
-The data flow between the layers is strictly limited to some few well-defined objects to ensure a clean separation of concerns. The presentation layer communicates with the service layer via a e.g. a record projection (i.e. a view of a record localised to a specific identity). The service layer communicates with the data access layer via e.g. a record entity that provides data abstraction, syntactic data validation, and a strong programmatic API.
+The data flow between the layers is strictly limited to some few well-defined objects to ensure a clean separation of concerns. The presentation layer communicates with the service layer via a record projection (i.e. a view of a record localised to a specific identity). The service layer communicates with the data access layer via a record entity that provides data abstraction, syntactic data validation, and a strong programmatic API.
 
 !!! tip "Tip: Where do you belong?"
```
Comment (fenekku) [inline docs/develop/architecture/software.md:36]: Just below is another fix to do: "where you code belongs" -> "where your code belongs"

File: docs/develop/getting-started/virtualenvs.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ programs (i.e. a distribution provides the ``python`` command).
 ### Distribution version
 
 Each distribution comes in multiple versions that's able to interpret the
-Python programming language - e.g Python 3.8, 3.9 etc.
+Python programming language - e.g., Python 3.8 and 3.9
```
Comment (fenekku) [inline docs/develop/getting-started/virtualenvs.md:27]: `Each distribution supports multiple version of the Python programming language e.g., Python 3.8 and Python 3.9`

(can't create a suggestion for unmodified line, but let's reword it too)

File: docs/develop/getting-started/virtualenvs.md [hunk 5]
```diff
@@ -57,15 +57,15 @@ Thus in addition to having multiple ``python`` commands, you can have multiple
...
 in a single Python distribution.
 
-Virtual environments allows you to have a virtual environment for program A and
-another for program B each with different version installed of package X.
+Virtual environments allow you to have a virtual environment for program A and
+another for program B each with a different version installed of package X.
 
-This means that a virtual environment has it's own ``python`` and ``pip``
+This means that a virtual environment has its own ``python`` and ``pip``
 commands that are *enhanced aliases* to the distributions ``python`` and ``pip``
```
Comment (fenekku) [inline docs/develop/getting-started/virtualenvs.md:69]: Don't know too much about that *enhanced aliases* business to be honest... I think cutting it at:

```
This means that a virtual environment has its own ``python`` and ``pip`` commands.
```

is good enough.

File: docs/maintenance/release-management.md [hunk 6]
```diff
@@ -86,7 +86,7 @@ For modules in ``vX.Y.Z``, the new version is ``v(X+1).0.0.dev0``.
 - Release Invenio-App-RDM (removing the pre-release suffix - e.g. ``dev0``).
 - Cookiecutter-Invenio-RDM:
     - Merge everything to ``master``.
-    - Create new version branch from ``master`` using the pattern ``vX.Y`` (e.g if Invenio-App-RDM is v1.0.0, the branch should be named ``v1.0``).
+    - Create a new version branch from ``master`` using the pattern ``vX.Y`` (e.g. if Invenio-App-RDM is v1.0.0, then the branch should be named ``v1.0``).
```
Comment (fenekku) [inline docs/maintenance/release-management.md:89]:
```suggestion
    - Create a new version branch from ``master`` using the pattern ``vX.Y`` (e.g., if Invenio-App-RDM is v1.0.0, then the branch should be named ``v1.0``).
```

**Comment by J4bbi** [comment]
@fenekku , further comments addressed

**Comment by fenekku** [comment]
image
